### PR TITLE
[EN-87] Displayed only Active employees by default

### PR DESCRIPTION
**Pull request by baremarco**
# Description :pencil:

we should display only employees that are active in the company.  However we still need the option to see all the people that is or was employee. To do so a new field called active was added to the DB
## Link to ticket :link:

https://entropyteam.atlassian.net/browse/EN-

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? :microscope:

- Verified the new column `active` on the local DB

### CODE REVIEW

File: src/main/java/com/entropyteam/entropay/employees/services/EmployeeService.java [hunk 1]
```diff
@@ -87,6 +87,7 @@ protected Employee toEntity(EmployeeDto dto) {
         employee.setTechnologies(technologies);
         employee.setPaymentsInformation(dto.paymentInformation() == null ? Collections.emptySet()
                 : dto.paymentInformation().stream().map(PaymentInformation::new).collect(Collectors.toSet()));
+        employee.setActive(true);
```
Comment (FacuOrellana) [inline src/main/java/com/entropyteam/entropay/employees/services/EmployeeService.java:90]: all the employees that are created will be active? @agustinb4 . Will need a criteria to set true in active field and query to verify the rest of employees active value

**Review by FacuOrellana — CHANGES_REQUESTED** [review]

File: src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java [hunk 2]
```diff
@@ -51,7 +51,8 @@ public record EmployeeDto(UUID id,
                           String client,
                           String role,
                           UUID lastAssignmentId,
-                          @JsonFormat(pattern = "yyyy-MM-dd") LocalDate startDate) {
+                          @JsonFormat(pattern = "yyyy-MM-dd") LocalDate startDate,
+                          @NotNull(message = "Active state is mandatory") boolean active) {
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java:55]: I would format this like:
```
@NotNull(message = "Active state is mandatory") 
boolean active
```

File: src/test/java/com/entropyteam/entropay/employees/services/ContractServiceTest.java [hunk 3]
```diff
@@ -78,6 +79,9 @@ class ContractServiceTest {
     private ContractService contractService;
     private Contract existentContract;
     private Contract activeContract;
+    
+    @Autowired
+    private EmployeeService employeeService;
```
Comment (agustinb4) [inline src/test/java/com/entropyteam/entropay/employees/services/ContractServiceTest.java:84]: I think you can remove the changes on this test file

**Review by agustinb4 — CHANGES_REQUESTED** [review]

File: src/main/java/com/entropyteam/entropay/employees/services/EmployeeService.java (hunk 1 above)
Comment (baremarco) [reply src/main/java/com/entropyteam/entropay/employees/services/EmployeeService.java:90]: Added active switch to address this
image

File: src/test/java/com/entropyteam/entropay/employees/services/ContractServiceTest.java [hunk 4]
```diff
@@ -35,6 +35,7 @@
 import org.mockito.Mock;
 import org.mockito.Spy;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (agustinb4) [inline src/test/java/com/entropyteam/entropay/employees/services/ContractServiceTest.java:38]: minor: please remove this import

File: src/main/resources/db/migration/V29.0__DDL_update_employee_table_update_active.sql
Comment (agustinb4) [inline src/main/resources/db/migration/V29.0__DDL_update_employee_table_update_active.sql:1]: @FacuOrellana @baremarco, there is a 29 migration in both pr's please sync before merge

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java [hunk 5]
```diff
@@ -257,4 +259,13 @@ public String getLabourEmail() {
     public void setLabourEmail(String labourEmail) {
         this.labourEmail = labourEmail;
     }
+
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/models/Employee.java:262]: minor: we have a wired format here

File: src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java [hunk 6]
```diff
@@ -51,7 +51,9 @@ public record EmployeeDto(UUID id,
                           String client,
                           String role,
                           UUID lastAssignmentId,
-                          @JsonFormat(pattern = "yyyy-MM-dd") LocalDate startDate) {
+                          @JsonFormat(pattern = "yyyy-MM-dd") LocalDate startDate,
+                          @NotNull(message = "Active state is mandatory") 
+						  boolean active) {
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java:56]: minor: we have a wired format here

**Review by agustinb4 — APPROVED** [review]

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java (hunk 5 above)
Comment (baremarco) [reply src/main/java/com/entropyteam/entropay/employees/models/Employee.java:262]: Do you mean the line break on line 262 ? May I have to remove it ?

File: src/main/resources/db/migration/V29.0__DDL_update_employee_table_update_active.sql
Comment (baremarco) [reply src/main/resources/db/migration/V29.0__DDL_update_employee_table_update_active.sql:1]: My migration updated to 30

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java (hunk 5 above)
Comment (agustinb4) [reply src/main/java/com/entropyteam/entropay/employees/models/Employee.java:262]: sorry, i mean this: 263 to 269 should be aligned to previous lines
format

**Review by agustinb4 — APPROVED** [review]
All right, just a minor comment

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java (hunk 5 above)
Comment (baremarco) [reply src/main/java/com/entropyteam/entropay/employees/models/Employee.java:265]: Ready :+1:

**Pull request merged**